src/components/FolderPage/FolderList.js

```diff
@@ -0,0 +1,61 @@
+import useFolderList from "../../hooks/useFolderList.js";
+import "./FolderList.css";
+import CardList from "../CardList.js";
+import { getFolderLinks } from "../../api.js";
+import { useState, useEffect } from "react";
+import UtilIcons from "./UtilIcons.js";
+import add from "../../assets/add.svg";
+import FolderButtons from "./FolderButtons";
+
+const FolderList = () => {
+  const folderList = useFolderList();
+  const [items, setItems] = useState([]);
+  const [selectedFolderName, setSelectedFolderName] = useState([]);
+
+  const handleLoadMockData = async () => {
+    const { data } = await getFolderLinks();
+    setItems(data);
+  };
+
+  useEffect(() => {
+    handleLoadMockData();
+  }, []);
+
+  const handleFolderClick = (folderId) => {
+    if (!folderId) {
+      handleLoadMockData();
+      setSelectedFolderName([]);
+    } else {
+      setSelectedFolderName(
+        folderList.find((folder) => folder.id === folderId)?.name
+      );
+      // assume JSON provides necessary data in folder.links
+      setItems(folderList.find((folder) => folder.id === folderId)?.links);
+    }
+  };
+
+  return (
+    <div className="wrapper">
+      <div className="folderListWrapper">
+        <div className="folderList">
+          {FolderButtons(folderList, handleFolderClick)}

```

Comment:
리액트 컴포넌트를 사용할때 함수가 아니라 컴포넌트로써 사용을 해주세요

```suggestion
          <FolderButtons propsName1={folderList} propsName2={handleFolderClick} />

```